test/csuite/wt4803_cache_overflow_abort/main.c

```diff
@@ -0,0 +1,116 @@
+#include "test_util.h"
+
+#include <signal.h>
+#include <sys/wait.h>
+
+#define	CONFIG_SIZE	100
+
+static int
+las_workload(TEST_OPTS *opts, const char *las_file_max)
+{
+	WT_CURSOR *cursor;
+	WT_SESSION *other_session, *session;
+	int i;
+	char buf[WT_MEGABYTE], open_config[CONFIG_SIZE];
+
+	WT_RET(__wt_snprintf(open_config, CONFIG_SIZE,
+	    "create,cache_size=50MB,cache_overflow=(file_max=%s)",
+	    las_file_max));
+
+	WT_RET(wiredtiger_open(opts->home, NULL, open_config, &opts->conn));
```

Comment:
That's correct. The test was only checking that we terminated with `SIGABRT`. Sorry, I got a bit confused about what you were suggesting initially since some of the tests also have an actual signal handler.

I've now made it so that if we get that particular panic message, I exit with a sentinel value of `EINVAL`.

Also, I had a think about this and we can avoid having this sentinel value and do the checking in the child process if we want to. So in the error handler we can have:

```
/* Some global variable that we can assign to in between tests. */
if (expect_panic)
    exit(EXIT_SUCCESS);
```

And a similar conditional for the case where we succeed but were expecting a panic. So the child will return success on a panic if we were expecting that to happen and vice versa. It seemed a bit icky to me so I went with the sentinel value approach but if you think this way is better, it shouldn't be hard to change it.

Reply:
Yes, I prefer your suggestion of having a global variable that the parent sets before each test and that the child can look for and exit with success when the right things happen on each run. Clearly expecting a panic should also look for the specific message we want. But yes, please go with that idea as I think it simplifies the program logic.

Reply:
98d8d9d